Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -20,30 +22,20 @@ async function run() {
 async function getJava(versionSpec: string) {
     const fromAzure: boolean = ('AzureStorage' == taskLib.getInput('jdkSourceOption', true));
     const extractLocation: string = taskLib.getPathInput('jdkDestinationDirectory', true);
-    const cleanDestinationDirectory: boolean = taskLib.getBoolInput('cleanDestinationDirectory', false);
+    const cleanDestDir: boolean = taskLib.getBoolInput('cleanDestinationDirectory', false);
     let compressedFileExtension: string;
     let jdkDirectory: string;
 
     toolLib.debug('Trying to get tool from local cache first');
     const localVersions: string[] = toolLib.findLocalToolVersions('Java');
     const version: string = toolLib.evaluateVersions(localVersions, versionSpec);
 
-     // Clean the destination folder before downloading and extracting?
-     if (cleanDestinationDirectory && taskLib.exist(extractLocation) && taskLib.stats(extractLocation).isDirectory) {
-        console.log(taskLib.loc('CleanDestDir', extractLocation));
-
-        // delete the contents of the destination directory but leave the directory in place
-        fs.readdirSync(extractLocation)
-        .forEach((item: string) => {
-            const itemPath = path.join(extractLocation, item);
-            taskLib.rmRF(itemPath);
-        });
-    }
-
     if (version) { //This version of Java JDK is already in the cache. Use it instead of downloading again.
         console.log(taskLib.loc('Info_ResolvedToolFromCache', version));
-    } else if (fromAzure) { //Download JDK from an Azure blob storage location and extract.
+    } else if ('AzureStorage' == taskLib.getInput('jdkSourceOption', true)) { //Download JDK from an Azure blob storage location and extract.
         console.log(taskLib.loc('RetrievingJdkFromAzure'));
+        cleanDestinationDirectory(cleanDestDir, extractLocation);
```

Comment:
I feel it's a bit non-standard to pass in the boolean to check to see if the method should even do anything.  I would suggest either changing the name of the method to something like: `checkCleanDestinationDirectory`, or put the if-check before calling the method.

Reply:
In thinking about it, I prefer the name `cleanDestDirIf` better (it fits with the `argIf` pattern used in vsts-task-lib/tools), but I'll leave the matter to your discretion.